### PR TITLE
Domain resource

### PR DESCRIPTION
**Pull request by akartsky**
Domain resource 

Take a look at following files :
 - generator.yaml
 - hook.go
 - template/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: helm/templates/deployment.yaml [hunk 1]
```diff
@@ -61,6 +61,8 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
+        - name: AWS_ACCOUNT_ID
```
Comment (ryansteakley) [inline helm/templates/deployment.yaml:64]: This should be removed.

File: helm/values.yaml [hunk 2]
```diff
@@ -38,6 +38,7 @@ resources:
 aws:
   # If specified, use the AWS region for AWS API calls
   region: ""
+  account_id: ""
```
Comment (ryansteakley) [inline helm/values.yaml:41]: this should be removed

File: pkg/resource/domain/hook.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+// requeueUntilCanModify creates and returns an
+// ackrequeue error if a resource's latest status matches
+// any of the defined modifying statuses below.
+func (rm *resourceManager) requeueUntilCanModify(ctx context.Context, r *resource) error {
+	latestStatus := r.ko.Status.Status
+	return svccommon.RequeueIfModifying(latestStatus, &resourceName, &modifyingStatuses)
+}
+
+// customDescribeDomainSetOutput sets the resource ResourceSynced condition to False if endpoint is
```
Comment (ryansteakley) [inline pkg/resource/domain/hook.go:49]: Is this supposed to say endpoint or domain?

File: helm/templates/deployment.yaml (hunk 1 above)
Comment (akartsky) [reply helm/templates/deployment.yaml]: fixed 
generated code with codegen commit 4b30ff557

File: helm/values.yaml (hunk 2 above)
Comment (akartsky) [reply helm/values.yaml]: done

File: pkg/resource/domain/hook.go (hunk 3 above)
Comment (akartsky) [reply pkg/resource/domain/hook.go]: done

**Review by ryansteakley — APPROVED** [review]

File: apis/v1alpha1/generator.yaml [hunk 4]
```diff
@@ -713,7 +713,49 @@ resources:
...
+        template_path: common/sdk_update_pre_build_request.go.tpl
+      dk_delete_pre_build_request:
+        template_path: common/sdk_delete_pre_build_request.go.tpl
+      sdk_delete_post_request:
+        template_path: common/sdk_delete_post_request.go.tpl
+    fields:
+      DomainId:
+        is_read_only: true
+        print:
+          name: DOMAIN-ID
```
Comment (mbaijal) [inline apis/v1alpha1/generator.yaml:742]: We have generally not printed columns other than status, any reason for adding this here ?

**Review by mbaijal — APPROVED** [review]

File: pkg/resource/domain/resource.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+// SetStatus will set the Status field for the resource
+func (r *resource) SetStatus(desired acktypes.AWSResource) {
+	r.ko.Status = desired.(*resource).ko.Status
+}
+
+// SetIdentifiers sets the Spec or Status field that is referenced as the unique
+// resource identifier
+func (r *resource) SetIdentifiers(identifier *ackv1alpha1.AWSIdentifiers) error {
```
Comment (mbaijal) [inline pkg/resource/domain/resource.go:94]: Verified that it looks correct.

**Pull request merged**